File: src/pewmode/Ship.java
package pewmode;

import org.newdawn.slick.Image;
import org.newdawn.slick.SlickException;


public class Ship extends Sprite {
    public Ship() throws SlickException {
        this.image = new Image("ship.png");
        init();
    }

    public void render() {
        image.draw(xPos, yPos, scale);
    }

    @Override
    public void move(float x, float y) {
        if (xPos > (70 - x) && xPos < (1740 - x)) {
            xPos += x;
        }
        if (yPos > (-20  - y) && yPos < (1110 - y)) {
            yPos += y;
        }
    }
}

File: src/pewmode/Ship.kt
package pewmode

import org.newdawn.slick.Image
import org.newdawn.slick.SlickException

class Ship
constructor() : Sprite() {
    init {
        image = Image("ship.png")
    }

    override fun render() {
        image.draw(xPos, yPos, scale)
    }

    override fun move(x: Float, y: Float) {
        if (xPos > (70 - x) && xPos < (1740 - x)) {
            xPos += x
        }

        if (yPos > (-20 - y) && yPos < (990 - y)) {
            yPos += y
        }
    }
}
